src/lib/krb5/krb/sendauth.c

```diff
@@ -137,13 +137,13 @@ krb5_sendauth(krb5_context context, krb5_auth_context *auth_context,
         d.length = sizeof (rnd_data);
         d.data = rnd_data;
         len2 = sizeof (rnd_data);
-        if (getpeername (*(int*)fd, (GETPEERNAME_ARG2_TYPE *) rnd_data,
+        if (getpeername (*(int*)fd, (GETPEERNAME_ARG2_TYPE *)(void *)rnd_data,
```

Comment:
Added a new fix removing the _ARG2_TYPE macros.  I think if for some reason we want to keep them we could also just declare `GETPEERNAME_ARG2_TYPE rnd_data;` and it would work.